### PR TITLE
changes for milestone 3.

### PR DESCRIPTION
**Pull request by ngomez-uy**
## Description

Changes added for Milestone 3. One challenge faced was that the import of the newly created Model failed at runtime. After a review with Juan, he told me we have to use the .js extension. I think, for someone that is learning, this is counter-intuitive because those are .ts files, not .js files, so it is a little bit confusing in my opinion. 

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Run Automation Tests:

Automation:

N/A

### CODE REVIEW

File: framework/example.env
Comment (jfagundez-uy) [inline framework/example.env:1]: Please copy the example.env to a .env file and remove the values from example.env. You are exposing sensitive data like login credentials.

File: framework/src/tests/PlaceOrder.spec.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import { OrderModel, OrderStatus } from "../models/request/OrderModel.js";
+import { StoreService } from "../models/services/StoreService.js";
+import { OrderResponse } from "../models/responses/OrderResponse.js";
+
+chai.should();
+
+describe("Add Order", () => {
+  const storeService = new StoreService();
+
```
Comment (jfagundez-uy) [inline framework/src/tests/PlaceOrder.spec.ts:10]: Remove extra line

File: framework/src/tests/PlaceOrder.spec.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    
+
+    response.status.should.equal(200, JSON.stringify(response.data));
+    response.data.id.should.be.above(0); //to verify it is a valid id
+    response.data.petId.should.be.above(0); //to verify it is a valid petId
+    response.data.quantity.should.be.above(0); //to verify it is a valid quantity
+    response.data.quantity.should.equal(order.quantity); //to verify it is the same quantity
+    formattedShipDate.should.equal(order.shipDate); //to verify it is the same shipDate
+    response.data.status.should.equal(order.status); //to verify it is the same status
+    response.data.complete.should.equal(order.complete); //to verify it is the same complete
```
Comment (jfagundez-uy) [inline framework/src/tests/PlaceOrder.spec.ts]: These comments seem unnecessary as they describe straightforward code or repeat what the code already clearly conveys. Over-commenting can reduce readability and maintenance ease. Comments should add context, explain ‘why’ decisions were made, or clarify complex logic—not restate ‘what’ the code does. I recommend removing these redundant comments to keep the code clean and focused

File: framework/src/tests/PlaceOrder.spec.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      shipDate: "2025-01-15",
+      status: OrderStatus.Placed,
+      complete: true,
+    };
+
+    const response = await storeService.placeOrder<OrderResponse>(order);
+    const formattedShipDate = response.data.shipDate.toString().split('T')[0];
+    if (!formattedShipDate) throw new Error("Ship Date is not defined");
+    
+
```
Comment (jfagundez-uy) [inline framework/src/tests/PlaceOrder.spec.ts]: Remove extra line

File: framework/src/tests/PlaceOrder.spec.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    response.status.should.equal(200, JSON.stringify(response.data));
+    response.data.id.should.be.above(0); //to verify it is a valid id
+    response.data.petId.should.be.above(0); //to verify it is a valid petId
+    response.data.quantity.should.be.above(0); //to verify it is a valid quantity
+    response.data.quantity.should.equal(order.quantity); //to verify it is the same quantity
+    formattedShipDate.should.equal(order.shipDate); //to verify it is the same shipDate
+    response.data.status.should.equal(order.status); //to verify it is the same status
+    response.data.complete.should.equal(order.complete); //to verify it is the same complete
+    console.log(formattedShipDate);
```
Comment (jfagundez-uy) [inline framework/src/tests/PlaceOrder.spec.ts]: Unnecessary console.log statements can reduce readability. Please remove any that aren’t essential for debugging or functionality

File: framework/src/tests/PlaceOrder.spec.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    
+    
+
```
Comment (jfagundez-uy) [inline framework/src/tests/PlaceOrder.spec.ts]: Remove extra lines

**Review by jfagundez-uy — CHANGES_REQUESTED** [review]
The PR looks good overall. However, the lib folder should be removed as it’s unnecessary and adds repetitive code. I also recommend adding invalid test cases to improve coverage and handle edge cases

**Comment by damianpereira86** [comment]
> The PR looks good overall. However, the lib folder should be removed as it’s unnecessary and adds repetitive code. I also recommend adding invalid test cases to improve coverage and handle edge cases

@jfagundez-uy Regarding adding tests, this milestone specifically asks for the creation only of the first test. Additional tests will be covered later.

File: framework/example.env
Comment (ngomez-uy) [reply framework/example.env:1]: I can remove them of course, but I added the example.env file to the .gitignore file. Shouldn't this be enough in order to not expose this sensitive info ?

Comment (jfagundez-uy) [reply framework/example.env:1]: We need to keep example.env versioned as it serves as a reference for the required environment variables. To avoid exposing sensitive information, the file should only contain placeholders or non-sensitive defaults. The sensitive values should be in .env file which is not tracked by git.

Comment (ngomez-uy) [reply framework/example.env:1]: So, I should take out "example.env" from the .gitignore, right? But leave the values blank of course in order to not expose sensitive info.

**Review by jfagundez-uy — APPROVED** [review]

**Review by damianpereira86 — APPROVED** [review]
Ready to be merged
